app/flatpak-builtins-kill.c

```diff
@@ -53,7 +53,8 @@ kill_instance (const char *id,
   for (j = 0; j < instances->len; j++)
     {
       FlatpakInstance *instance = (FlatpakInstance *) g_ptr_array_index (instances, j);
-      if (strcmp (id, flatpak_instance_get_app (instance)) == 0 ||
+      const char *app_name = flatpak_instance_get_app(instance);
+      if ((app_name && strcmp (id, flatpak_instance_get_app (instance)) == 0) ||
```

Comment:
```suggestion
      if (g_strcmp0 (id, flatpak_instance_get_app (instance)) == 0 ||
```

Reply:
I committed this change for you, hope you don't mind. Thanks for the patch!

Reply:
Thanks! This is now definitely more complete.